apps/devhub/views.py

```diff
@@ -1735,3 +1729,26 @@ def docs(request, doc_name=None):
 def search(request):
     query = request.GET.get('q', '')
     return render(request, 'devhub/devhub_search.html', {'query': query})
+
+
+def api_key_agreement(request):
+    next_step = reverse('devhub.api_key')
+    return render_agreement(request, 'devhub/api/agreement.html', next_step)
+
+
+def render_agreement(request, template, next_step, step=None):
+    if request.method == 'POST':
+        request.user.update(read_dev_agreement=now())
+        return redirect(next_step)
+
+    if request.user.read_dev_agreement is None:
+        return render(request, template,
+                      {'step': step})
+    else:
+        response = redirect(next_step)
```

Comment:
This might need some improvement in the future: a dev uploading a new add-on will get redirected straight to step2 if he already accepted the agreement. Clicking on the big fat unread link "1. Getting started" will have no effect (other than redirecting straight to the same step 2).

I believe it would make more sense to have this "step 1" taken out of the flow, and just be an extra initial requirement (deliberately not talking about a "step" here) to the step 1 (which is uploading the add-on).